fe/fe-core/src/main/java/com/starrocks/analysis/Parameter.java

```diff
@@ -55,10 +58,17 @@ public <R, C> R accept(AstVisitor<R, C> visitor, C context) {
 
     @Override
     public Type getType() {
+        Type res;
         if (expr != null) {
-            return expr.getType();
+            res = expr.getType();
+        } else {
+            res = super.getType();
         }
-        return super.getType();
+        // use STRING as default type, since string is compatible with almost all types in the type inference
```

Comment:
When will the type be Invalid?

Reply:
I'm a little worried that we can't get invalid type when this is really needed. For example if the SQL is not correct.

Reply:
The type of `Parameter` is INVALID, bc we don't need its type when executing `PREPARE ` statement.

The type would be determined in the `EXECUTE` statement, in which the parameter is provided and we the inference the type bottomup